src/figuresClasses.ts

```diff
@@ -1,19 +1,73 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: string,
+  color: string,
+  getArea: () => number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c].sort((n1, n2) => n2 - n1);
+    const isPositiveSides = sides.some((el) => el <= 0);
+    const isInvalidSide = sides[0] >= sides[1] + sides[2];
+
+    if (isPositiveSides || isInvalidSide) {
+      throw new Error(`Fucking ${this.shape} Error`);
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const s = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
 
+    return Math.floor(s * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error(`Fucking ${this.shape} Error`);
+    }
+  }
+
+  getArea(): number {
+    return Math.floor((Math.PI * this.radius ** 2) * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0) {
+      throw new Error(`Fucking ${this.shape} Error`);
+    }
```

Comment:
you checked is all input arguments are bigger than 0 for each figure
let's create several function for this purpose, that will accept the arguments and will check if each of them is bigger than 0
and will call one function for all figures